src/core/tiledmesh/qgscesiumtilesdataprovider.cpp

```diff
@@ -0,0 +1,183 @@
+/***************************************************************************
+                         qgscesiumtilesdataprovider.cpp
+                         --------------------
+    begin                : June 2023
+    copyright            : (C) 2023 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgscesiumtilesdataprovider.h"
+#include "qgsproviderutils.h"
+#include "qgsapplication.h"
+#include "qgsprovidersublayerdetails.h"
+#include "qgsthreadingutils.h"
+
+#include <QUrl>
+#include <QIcon>
+
+///@cond PRIVATE
+
+#define PROVIDER_KEY QStringLiteral( "cesium" )
+#define PROVIDER_DESCRIPTION QStringLiteral( "Cesium 3D Tiles data provider" )
+
+QgsCesiumTilesDataProvider::QgsCesiumTilesDataProvider( const QString &uri, const ProviderOptions &providerOptions, ReadFlags flags )
+  : QgsTiledMeshDataProvider( uri, providerOptions, flags )
+{
+
+}
+
+QgsCesiumTilesDataProvider::~QgsCesiumTilesDataProvider() = default;
+
+QgsCoordinateReferenceSystem QgsCesiumTilesDataProvider::crs() const
+{
+  QGIS_PROTECT_QOBJECT_THREAD_ACCESS
+
+  // TODO
+  return QgsCoordinateReferenceSystem();
+}
+
+QgsRectangle QgsCesiumTilesDataProvider::extent() const
+{
+  QGIS_PROTECT_QOBJECT_THREAD_ACCESS
+
+  // TODO
+  return QgsRectangle();
+}
+
+bool QgsCesiumTilesDataProvider::isValid() const
+{
+  QGIS_PROTECT_QOBJECT_THREAD_ACCESS
+
+  // TODO
+  return true;
+}
+
+QString QgsCesiumTilesDataProvider::name() const
+{
+  QGIS_PROTECT_QOBJECT_THREAD_ACCESS
+
+  return PROVIDER_KEY;
+}
+
+QString QgsCesiumTilesDataProvider::description() const
+{
+  QGIS_PROTECT_QOBJECT_THREAD_ACCESS
+
+  return QObject::tr( "Cesium 3D Tiles" );
+}
+
+//
+// QgsCesiumTilesProviderMetadata
+//
+
+QgsCesiumTilesProviderMetadata::QgsCesiumTilesProviderMetadata():
+  QgsProviderMetadata( PROVIDER_KEY, PROVIDER_DESCRIPTION )
+{
+}
+
+QIcon QgsCesiumTilesProviderMetadata::icon() const
+{
+  return QgsApplication::getThemeIcon( QStringLiteral( "xxxcesiumxxx.svg" ) );
```

Comment:
Actual icon coming in a separate PR?

Reply:
Correct :wink: .. my current attempts at making a nice looking 3d tiles icon at the required small size are failing badly